### PR TITLE
feat: add 11 L3/L4/L5-only ChangeKinds and worked example cases

### PR DESCRIPTION
**Pull request by napetrov**
## Summary

Add 11 build-context (L3), source-replay (L4), and source-graph (L5) `ChangeKind`s for ABI/API failures that no artifact layer (L0/L1/L2) can observe, plus a compiler-free example corpus (cases 152–162) demonstrating each.

## Motivation

The evidence-layer model (`docs/concepts/evidence-and-detectability.md`) is explicit that some breaks are structurally invisible to artifact comparison — they live in the flags a library was *built* with (L3), in source constructs that never reach the binary (L4 macros, inline bodies, `constexpr`), or in derived reachability/ownership deltas (L5). This fills concrete gaps in that space with dedicated, actionable findings instead of the generic `abi_relevant_build_flag_changed` bucket or no finding at all.

## Changes

New `ChangeKind`s (enum 254 → 265). Per ADR-028 D3 none are `BREAKING` on their own — they default to `API_BREAK` (source breaks) or `RISK` (deployment/context risk); the artifact diff still proves any shipped-ABI break.

- **L3 build-context flag flips (RISK)** — `enum_size_flag_changed` (`-fshort-enums`), `struct_packing_mode_changed` (`-fpack-struct` / `/Zp`), `lto_mode_changed` (`-flto`), `char_signedness_changed` (`-fsigned-char` ↔ `-funsigned-char`, both sides must be explicit since the default is target-dependent). Normalized in `adapters/base.py`, routed to dedicated findings in `build_diff.py`.
- **L4 source-replay findings** — `public_macro_removed` (API_BREAK), `inline_function_removed` (API_BREAK), `public_typedef_removed` (API_BREAK), `constexpr_function_body_changed` (RISK). Removal passes + constexpr-body check in `source_diff.py`.
- **L5 version-over-version graph deltas (RISK)** — `public_api_internal_dependency_added`, `target_dependency_added`, `exported_symbol_source_owner_changed`. New helpers + findings in `source_graph.py`.
- **Example cases 152–162** — each ships a hand-built pair of evidence-model fixtures (`old.json`/`new.json`) instead of compiled binaries, so the corpus is validated in the fast lane with **no compiler/castxml** (`scripts/gen_l3l4l5_examples.py`, `tests/test_l3l4l5_examples.py`), plus a README per case.
- **Detector unit tests** — `tests/test_l3l4l5_new_kinds.py` (20 tests, fast-lane) exercise every new kind and assert its partition, including the no-op-suppression edge cases.
- **Registry + gates** — impact text in `change_registry.py`, evidence-tier map in `scripts/evidence_tiers.py`, regenerated `detector-spec`, extended L3/L4/L5 tables in `concepts/build-source-data.md` (linking each example), and all ChangeKind-count / catalog-size doc anchors updated (254→265 kinds, 152→163 cases). `test_changekind_completeness` and `test_validate_examples_unit` counts updated.

## Checklist

- [x] Tests added / updated for new behaviour (`test_l3l4l5_new_kinds.py`, `test_l3l4l5_examples.py`, completeness/count updates)
- [x] `CHANGELOG.md` updated (under `[Unreleased]`)
- [x] Docs updated (findings tables, example pages, count anchors)
- [x] No new `mypy` or `ruff` errors introduced (`ruff check` clean, `mypy` 0 issues, AI-readiness 0 errors, all `--check` gates in sync)

**Note:** `tests/test_build_query_inference.py::test_run_bazel_empty_action_graph_is_partial` fails in this environment, but it also fails on the base commit (verified via `git stash`) — it is pre-existing and unrelated to this change.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_01QfNodPG5p6y4uxkNBw8LQ8)_

### CODE REVIEW

File: abicheck/buildsource/source_graph.py [hunk 1]
```diff
@@ -838,6 +838,74 @@ def _option_symbol_edges(graph: SourceGraphSummary) -> set[tuple[str, str]]:
...
+    """
+    kinds = _kind_map(graph)
+    symbol_to_decls: dict[str, list[str]] = {}
+    for e in graph.edges:
+        if e.kind == "SOURCE_DECL_MAPS_TO_SYMBOL":
+            symbol_to_decls.setdefault(e.dst, []).append(e.src)
+    decl_to_files: dict[str, list[str]] = {}
+    for e in graph.edges:
+        if e.kind == "SOURCE_DECLARES" and kinds.get(e.src) in _DECLARING_FILE_KINDS:
+            decl_to_files.setdefault(e.dst, []).append(e.src)
```
[the remark this answers is not in the document]
Comment (napetrov) [reply abicheck/buildsource/source_graph.py:900]: Good catch on the semantic mismatch — but `def_file` can't be the source here. `SourceGraphSummary.add_node` is first-writer-wins ("de-duplicating by id (first writer wins on facts)"), and in `build_source_graph` the exported `source_decl` node is always created by the source-ABI pass (which emits `SOURCE_DECL_MAPS_TO_SYMBOL`) *before* `augment_graph_with_calls` runs. So the call-graph augmentation's `add_node(... attrs={"def_file": ...})` is dropped for exactly the exported decls this detector iterates — reading `def_file` would make the detector dead for exported symbols, which is the failure mode this review has (rightly) been catching elsewhere in the PR.

I've instead committed the detector to the signal that *is* computable on production graphs — the declaring file (the `SOURCE_DECLARES` header/source node) — and re-aligned the finding text and code comment to "declaring file moved" rather than "implementation TU moved". Detecting a true definition-TU relocation would need a real `SOURCE_DEFINES` edge emitted during graph construction (or merging `def_file` onto pre-existing nodes); that's a graph-schema change and a good follow-up, but out of scope for this PR.

---
_Generated by [Claude Code](https://claude.ai/code)_
